wgpu/src/backend/wgpu_core.rs

```diff
@@ -2991,6 +2997,77 @@ fn default_error_handler(err: crate::Error) {
     panic!("wgpu error: {err}\n");
 }
 
+fn span_to_source_span(span: &naga::Span) -> Option<SourceSpan> {
+    span.to_range().map(|range| SourceSpan {
+        offset: range.start as u32,
+        length: range.len() as u32,
+    })
+}
+
+impl From<&CreateShaderModuleError> for CompilationInfo {
+    fn from(value: &CreateShaderModuleError) -> Self {
+        let message = match value {
+            #[cfg(feature = "wgsl")]
+            CreateShaderModuleError::Parsing(e) => CompilationMessage {
+                message: e.to_string(),
+                message_type: CompilationMessageType::Error,
+                span: e
+                    .inner
+                    .labels()
+                    .next()
+                    .and_then(|(span, _)| span_to_source_span(&span)),
```

Comment:
I wasn't entirely sure if I should only take the first span, or do something different.